contrib/gin-gonic/gin/appsec.go

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package gin
+
+import (
+	"net"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation/httpsec"
+
+	"github.com/gin-gonic/gin"
+)
+
+func useAppSec(c *gin.Context) {
+	req := c.Request
+	span, ok := tracer.SpanFromContext(req.Context())
+	if ok {
+		httpsec.SetAppSecTags(span)
+		params := make(map[string]string)
+		for _, p := range c.Params {
+			params[p.Key] = p.Value
+		}
```

Comment:
Let's keep the params map nil when there's no params and otherwise preallocate it.
```suggestion
		var params map[string]string
		if l := len(c.Params); l > 0 {
			params = make(map[string]string, l)
			for _, p := range c.Params {
				params[p.Key] = p.Value
			}
		}
```